creator/ingest_runs/data_generator/study_generator.py

```diff
@@ -0,0 +1,504 @@
+        )
+        fp = os.path.join(self.data_dir, 's3_gf_manifest.tsv')
+        if not create:
+            return read_df(fp)
+
+        s3_data = fetch_aws_bucket_obj_info(
+            bucket_name=self.study_bucket,
+            search_prefixes=[
+                "source/genomic-files/",
+                "harmonized/genomic-files/"
+            ],
+        )
+        df = pd.DataFrame(s3_data)
+
+        # Filter out folders
+        df["Filename"] = df["Key"].map(lambda x: os.path.split(x)[-1])
+        df = df[~df["Filename"].isnull()]
+
+        df.to_csv(fp, sep="\t", index=False)
+        self.logger.info(f"Wrote S3 manifest for genomic files: {fp}")
+
+        return df
+
+    def _create_bix_gwo_manifest(self, dfs=None):
+        """
+        Create BIX genomic workflow output manifest
+        """
+        create = dfs
+        fp = os.path.join(self.data_dir, 'gwo_manifest.tsv')
+        verb = "Creating" if create else "Reading"
+        self.logger.info(
+            f"{verb} bioinformatics genomic workflow output manifest {fp}"
+        )
+        if not create:
+            return read_df(fp)
+
+        def create_hex(n):
+            """
+            Return a random hex number of length n.
+            """
+            return "".join(
+                [ra.choice(string.hexdigits).lower() for _ in range(n)]
+            )
+
+        def data_type(row):
+            if row["filepath"].endswith("g.vcf.gz"):
+                dt = "gVCF"
+            else:
+                dt = None
+            return dt
+
+        def workflow_type(row):
+            if row["Data Type"] in {"gVCF"}:
+                wt = "alignment"
+            else:
+                wt = None
+            return wt
+
+        # Merge all dfs
+        bio_df, gf_df, s3_df = dfs
+        s3_df.columns = [c.lower() for c in s3_df.columns]
+        df = pandas_utils.merge_wo_duplicates(bio_df, gf_df, on="sample_id")
+        df = pandas_utils.merge_wo_duplicates(df, s3_df, on="filepath")
+        # Add additional columns
+        df["Data Type"] = df.apply(lambda row: data_type(row), axis=1)
+        df["Workflow Type"] = df.apply(lambda row: workflow_type(row), axis=1)
+        df["Cavatica ID"] = create_hex(25)
+        df["Cavatica Task ID"] = (
+            "-".join([create_hex(i) for i in [8, 4, 4, 4, 12]])
+        )
+        # Rename columns
+        col_map = {
+            "kf_id_participant": "KF Participant ID",
+            "kf_id_biospecimen": "KF Biospecimen ID",
+            "kf_id_family": "KF Family ID",
+            "filepath": "Filepath",
+            "source_read": "Source Read",
+        }
+        df = df[
+            list(col_map.keys()) +
+            ["Data Type", "Workflow Type", "Cavatica ID", "Cavatica Task ID"]
+        ].rename(columns=col_map)
+
+        df.to_csv(fp, sep="\t", index=False)
+        self.logger.info(f"Wrote {fp}")
+
+        return df
+
+    def _assert_load(self, endpoint, kf_ids):
```

Comment:
Added a sanity check to make sure that all of the KF IDs we generated in the files were actually loaded into the Data Service